_posts/2020_04_29_encrypted_keras_with_syft/encrypted_keras_with_syft.Rmd

```diff
@@ -0,0 +1,413 @@
+countermeasures. Of course, in the end, we'd like to see them all integrated -- but here's a journey where people are just
+starting to embark. The most important thing we can do, then, is to learn about the concepts, sight the landscape of
+implementations under development, and -- perhaps -- decide to join the effort.
+
+This post tries to do a tiny little bit of all of those.
+
+## Aspects of privacy in deep learning
+
+Say you work at a hospital, and would be interested in training a deep learning model to help diagnose some disease from brain
+scans. Where you work, you don't have many patients with this disease; moreover, they tend to mostly be affected by the same
+subtypes: Your training set, were you to create one, would not very well reflect the overall distribution. It'd thus make
+sense to cooperate with other hospitals; but that isn't so easy, as the data collected is protected by privacy regulations.
+So, the first requirement is: The data has to stay where it is; e.g., it may not be sent to a central server.
+
+#### Federated learning
+
+This first *sine qua non* is addressed by [federated
+learning](https://blogs.rstudio.com/tensorflow/posts/2020-04-08-tf-federated-intro/) [@McMahanMRA16]. Federated learning is
+not "just" desirable for privacy reasons. On the contrary, in many use cases, it may be the only viable way (like with
+smartphones or sensors, which collect gigantic amounts of data). In federated learning, each participant receives a copy of
+the model, trains on their own data, and sends back the gradients obtained to the central server, where gradients are averaged
+and applied to the model.
+
+This is good insofar as the data never leaves the individual devices; however, a lot of information can still be extracted
+from plain-text gradients. Imagine a smartphone app that provides trainable auto-completion for text messages. Even if
+gradient updates from many iterations are averaged, their distributions will greatly vary between individuals. Some form of
+encryption is needed. But then how is the server going to make sense of the encrypted gradients?
+
+One way to accomplish this relies on *secure multi-party computation* (SMPC).
+
+#### Secure multi-party computation
+
+In SMPC, we need a system of several agents who collaborate to provide a result no single agent could provide alone: "normal"
+computations (like addition, multiplication ...) on "secret" (encrypted) data. The assumption is that these agents are "honest
+but curious" -- honest, because they won't tamper with their share of data; curious in the sense that if they *were* (curious,
+that is), they wouldn't be able to inspect the data because it's encrypted.
+
+The principle behind this is *secret sharing*. A single piece of data -- a salary, say -- is "split up" into meaningless
+(hence, encrypted) parts which, when put together again, yield the original data. Here is an example.
```

Comment:
Is meaningless (on its own) the definition of encryption you're working with? I think that privacy terminology is _super_ easy to mix up, so I might define your terms at first usage in a side- or footnote.

Reply:
I see you expand on it below, so feel free to stet this comment.